### ISSUE TITLE
Indexed flag for field

### ISSUE THREAD

**Issue by tumb1er** [rutube/alco#48]
When filtered field is marked as indexed, sphinxsearch index enables separate spx_field for this field. While filtering, lookup is done by full-text search within this sphinx field.

**Comment by tumb1er** [comment]
TL;DR

For in-memory indices full scan is quicklier (5sec vs 20 sec)

13M rows

``` sql
SET profiling=1;
SELECT 
  (SNIPPET(logline, '279840', 'limit=1000000')) AS `logline_snippet`, 
  id, logline, js, js.allowed_hosts, js.application, js.data, js.duration, 
  js.error, js.exc_info, js.handler_args, js.handler_kwargs, js.host, 
  js.host_stats, js.levelname, js.lineno, js.params, js.process, js.project, 
  js.results, js.session, js.site, js.sql, js.subsystem, js.tags, js.task_id, 
  js.task_name, js.threadName, js.type, 
  id >= 1059437444000000000 AS `__where_result` 
FROM duck_20160331 
WHERE 
  MATCH('@logger ("celery.evcam")') 
  AND __where_result=1  
ORDER BY id ASC 
LIMIT 0,101 
OPTION max_matches=200, ranker=none;
SHOW PROFILE;
```

| Status | Duration | Switches | Percent |
| --- | --- | --- | --- |
| unknown | 0.017984 | 77 | 0.09 |
| net_read | 0.000005 | 1 | 0.00 |
| local_search | 0.200530 | 183 | 0.96 |
| sql_parse | 0.000139 | 1 | 0.00 |
| dict_setup | 0.000088 | 37 | 0.00 |
| parse | 0.001765 | 37 | 0.01 |
| transforms | 0.000164 | 37 | 0.00 |
| init | 0.018002 | 146 | 0.09 |
| init_segment | 0.000226 | 23 | 0.00 |
| open | 0.001530 | 36 | 0.01 |
| read_docs | 4.501362 | 1551 | 21.45 |
| read_hits | 7.762670 | 5553 | 36.99 |
| get_docs | 4.257224 | 26246 | 20.28 |
| filter | 3.693087 | 19024 | 17.60 |
| sort | 0.501924 | 37791 | 2.39 |
| finalize | 0.029720 | 37 | 0.14 |
| aggregate | 0.000080 | 2 | 0.00 |
| net_write | 0.000034 | 1 | 0.00 |
| eval_post | 0.000039 | 102 | 0.00 |
| eval_snippet | 0.000670 | 101 | 0.00 |
| total | 20.987243 | 90986 | 0 |

``` sql
SET profiling=1;
SELECT 
  (SNIPPET(logline, '279840', 'limit=1000000')) AS `logline_snippet`, 
  id, logline, js, js.allowed_hosts, js.application, js.data, js.duration, 
  js.error, js.exc_info, js.handler_args, js.handler_kwargs, js.host, 
  js.host_stats, js.levelname, js.lineno, js.params, js.process, js.project, 
  js.results, js.session, js.site, js.sql, js.subsystem, js.tags, js.task_id, 
  js.task_name, js.threadName, js.type,
  (id >= 1059437444000000000 AND js.logger = 'celery.evcam') AS `__where_result` 
FROM duck_20160331 
  WHERE __where_result=1  
  ORDER BY id ASC 
  LIMIT 0,101 
  OPTION max_matches=200, ranker=none;
SHOW PROFILE;
```

| Status | Duration | Switches | Percent |
| --- | --- | --- | --- |
| unknown | 0.001199 | 5 | 0.02 |
| net_read | 0.000003 | 1 | 0.00 |
| local_search | 0.207155 | 3 | 3.96 |
| sql_parse | 0.000153 | 1 | 0.00 |
| fullscan | 4.992161 | 36 | 95.42 |
| dict_setup | 0.000043 | 1 | 0.00 |
| parse | 0.000077 | 1 | 0.00 |
| transforms | 0.000004 | 1 | 0.00 |
| init | 0.000444 | 2 | 0.01 |
| finalize | 0.029508 | 37 | 0.56 |
| aggregate | 0.000080 | 2 | 0.00 |
| net_write | 0.000033 | 1 | 0.00 |
| eval_post | 0.000039 | 102 | 0.00 |
| eval_snippet | 0.000676 | 101 | 0.01 |
| total | 5.231575 | 294 | 0 |

**Comment by tumb1er** [comment]
Same data, but 60M rows

Full Scan:

| Status | Duration | Switches | Percent |
| --- | --- | --- | --- |
| unknown | 0.001183 | 5 | 0.00 |
| net_read | 0.000004 | 1 | 0.00 |
| local_search | 0.370140 | 3 | 0.58 |
| sql_parse | 0.000123 | 1 | 0.00 |
| fullscan | 63.818072 | 135 | 99.28 |
| dict_setup | 0.000042 | 1 | 0.00 |
| parse | 0.000064 | 1 | 0.00 |
| transforms | 0.000005 | 1 | 0.00 |
| init | 0.000388 | 2 | 0.00 |
| finalize | 0.085395 | 136 | 0.13 |
| aggregate | 0.000059 | 2 | 0.00 |
| net_write | 0.001380 | 1 | 0.00 |
| eval_post | 0.000019 | 102 | 0.00 |
| eval_snippet | 0.002252 | 101 | 0.00 |
| total | 64.279126 | 492 | 0 |

Match(column):

| Status | Duration | Switches | Percent |
| --- | --- | --- | --- |
| unknown | 0.034486 | 275 | 0.08 |
| net_read | 0.000003 | 1 | 0.00 |
| local_search | 0.940708 | 666 | 2.26 |
| sql_parse | 0.000090 | 1 | 0.00 |
| dict_setup | 0.000238 | 136 | 0.00 |
| parse | 0.005921 | 136 | 0.01 |
| transforms | 0.000370 | 136 | 0.00 |
| init | 0.065942 | 531 | 0.16 |
| init_segment | 0.000327 | 26 | 0.00 |
| open | 0.006217 | 135 | 0.01 |
| read_docs | 2.201880 | 6153 | 5.29 |
| get_docs | 15.328949 | 106603 | 36.83 |
| filter | 19.880668 | 100139 | 47.77 |
| sort | 3.068974 | 198915 | 7.37 |
| finalize | 0.080893 | 136 | 0.19 |
| aggregate | 0.000054 | 2 | 0.00 |
| net_write | 0.001220 | 1 | 0.00 |
| eval_post | 0.000030 | 102 | 0.00 |
| eval_snippet | 0.000548 | 101 | 0.00 |
| total | 41.617518 | 414195 | 0 |
